### PR TITLE
feat: Add temporal entity dimension filtering to dashboard filters

### LINKED ISSUE #1555: Check if temporal entity dimension work with dashboard shared filter
> Wondering if `TemporalEntityDimension` wouldn't also work "out of the box", as it looks like it has a `timeUnit` just like `TemporalDimension`?

_Originally posted by @bprusinowski in https://github.com/visualize-admin/visualization-tool/pull/1545#discussion_r1624444483_

### PR DESCRIPTION
**Pull request by bprusinowski**
...and fix the time range brush for `TemporalEntityDimension` in column charts.

Closes #1555

### Time brush for temporal entity dimension based X axis
**PR**
1. Go to [this link](https://visualization-tool-git-fix-temporal-entity-filtering-ixt1.vercel.app/en/create/new?cube=https://agriculture.ld.admin.ch/foag/cube/MilkDairyProducts/Consumption_Price_Month&dataSource=Int).
2. Open Horizontal Axis settings.
3. Enable interactive filter.
4. ✅ See that it's possible to filter the chart.

**TEST**
1. Go to [this link](https://test.visualize.admin.ch/en/create/new?cube=https://agriculture.ld.admin.ch/foag/cube/MilkDairyProducts/Consumption_Price_Month&dataSource=Int).
2. Open Horizontal Axis settings.
3. Enable interactive filter.
4. ❌ See that it's not possible to filter the chart (NaN values displayed in the brush).

### Temporal entity dimension in shared dashboard filters
**PR**
1. Go to [this link](https://visualization-tool-git-fix-temporal-entity-filtering-ixt1.vercel.app/en/create/new?cube=https://agriculture.ld.admin.ch/foag/cube/MilkDairyProducts/Consumption_Price_Month&dataSource=Int).
2. Add another chart (e.g. line).
3. Proceed to layout options.
4. Switch to a dashboard layout.
5. Enable shared Date filter.
6. Filter the chart using the time brush.
7. ✅ See that both charts are filtered as expected.

**TEST**
1. Go to [this link](https://test.visualize.admin.ch/en/create/new?cube=https://agriculture.ld.admin.ch/foag/cube/MilkDairyProducts/Consumption_Price_Month&dataSource=Int).
2. Add another chart (e.g. line).
3. Proceed to layout options.
4. Switch to a dashboard layout.
5. ❌ See that it's not possible to enable shared Date filter.

### CODE REVIEW

File: app/configurator/components/layout-configurator.tsx [hunk 1]
```diff
@@ -290,8 +306,9 @@ const SharedFilterOptions = ({
   if (sharedFilter.type !== "timeRange") {
     return null;
   }
+
   if (
-    !isTemporalDimension(dimension) ||
+    !(isTemporalDimension(dimension) || isTemporalEntityDimension(dimension)) ||
```
Comment (ptbrowne) [inline app/configurator/components/layout-configurator.tsx:311]: I guess this check and the ones before could be refactored as a "canDimensionBeTimeFiltered" or something like this ?

File: app/configurator/components/ui-helpers.ts [hunk 2]
```diff
@@ -368,16 +369,29 @@ export const extractDataPickerOptionsFromDimension = ({
...
+            label: `${d.value}`,
+            value: `${d.value}`,
+          };
+        })
+      : values.map((d) => {
+          return {
+            label: `${d.label}`,
+            value: `${d.position}`,
+          };
+        });
```
Comment (ptbrowne) [inline app/configurator/components/ui-helpers.ts:390]: Nit: Wonder if we should use ts-pattern here for exhaustive check.

**Review by ptbrowne — APPROVED** [review]
👏 👏 👏

**Comment by ptbrowne** [comment]
I think we could remove the Slider ticks when there are more than 50 ? 

image

File: app/configurator/components/ui-helpers.ts (hunk 2 above)
Comment (bprusinowski) [reply app/configurator/components/ui-helpers.ts:398]: Yup 👍

**Pull request merged**